rustls/src/crypto/signer.rs

```diff
@@ -111,4 +118,16 @@ impl CertifiedKey {
             .first()
             .ok_or(Error::NoCertificatesPresented)
     }
+
+    /// Verify the consistency of this [`CertifiedKey`]'s public and private keys.
+    /// This is done by performing a comparison of SubjectPublicKeyInfo bytes.
+    pub fn certificate_and_key_are_consistent(&self) -> Result<bool, Error> {
+        Ok(
```

Comment:
This formulation is pretty dense. Even though we often avoid single-use bindings, I think it would be nice to structure this like:

```rust
let public_spki = ParsedCertificate::try_from(self.end_entity_cert()?)?.subject_public_key_info();
let private_spki = self.key.public_key()?.ok_or(..)?;
Ok(public_spki == private_spki)
```

Reply:
I rewrote this taking into account @cpu's advice in [b3577ab](https://github.com/rustls/rustls/pull/1954/commits/b3577abc6b66c77c0e5024adcd8248f8f70115f7). The suggestion above doesn't exactly work because a temporary value is dropped whilst being borrowed, but it should look more simple nonetheless